### ISSUE TITLE
Image.getSize not returning correct image dimension from url

### ISSUE THREAD

**Issue by divyaswormakai** [facebook/react-native#33498]
### Description

![image](https://user-images.githubusercontent.com/38316121/160110271-190738a5-c551-4bf2-b117-06731473bbee.png)
* Image is taken from 'react-native';

The above getSize method from react-native returns the minimum of dp of the screen and size of image(height/width)
For example, 
dp = 1920
image width = 800
The returned width is 800

If image width. = 3000
The returned width is 1920.

Shouldn't the getSize return the size of the image?

### Version

0.66.4

### Output of `npx react-native info`

System:
    OS: macOS 12.3
    CPU: (8) x64 Apple M1 Pro
    Memory: 25.67 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.5 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.14 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 32.0.0
      System Images: android-25 | Google APIs ARM EABI v7a, android-28 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom_64, android-Sv2 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8139111
    Xcode: 13.3/13E113 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.4 => 0.66.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Use a 1920x1080 resolution screen.
2. Grab a 4k or 8k image. 
3. Use the getSize method of Image to get the size of the image.

**Comment by Marcoo09** [comment]
@divyaswormakai could you provide example images in which that happen? 
I tested with [Image1](https://images.pexels.com/photos/842711/pexels-photo-842711.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260) and [Image2](https://www.xtrafondos.com/wallpapers/edificios-ciudad-de-noche-3183.jpg) and both works correctly

**Comment by divyaswormakai** [comment]
@Marcoo09 
`
Image.getSize(
		'https://images.pexels.com/photos/842711/pexels-photo-842711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2',
		(width, height) => {
			console.log(`Width:${width}, Height:${height}`);
		},
	);
`

I used the first Image that you have.
The original image dimension is
<img width="252" alt="image" src="https://user-images.githubusercontent.com/38316121/160763535-5fe3cebe-04d9-426b-a871-95b8c1faa059.png">

From the React Native app
<img width="213" alt="image" src="https://user-images.githubusercontent.com/38316121/160763595-a4bace73-991b-4c8b-986a-b7f710b266f3.png">

I am testing it in Xiaomi Poco F1 screen which has the following dimension
<img width="468" alt="image" src="https://user-images.githubusercontent.com/38316121/160764209-a997b553-ffa2-4450-9bdc-6385758a1939.png">

**Comment by Marcoo09** [comment]
@divyaswormakai it looks like a book on the fresco library used on android implementation. https://github.com/facebook/fresco/issues/2236
I don't know how to solve it but in some comments mention this library which has a different implementation than react native https://www.npmjs.com/package/react-native-image-size

**Comment by divyaswormakai** [comment]
Hi @Marcoo09,

Thank you for the reply. The package that you have mentioned does work but if you look into the library, they convert the image to a bitmap and then get the real image dimension from it. It works okay for small images but for large images, it takes too much processing time directly hampering the app flow.
I have addressed this in the git repo as well.
https://github.com/eXist-FraGGer/react-native-image-size/issues/13

Also thank you for reopening the issue on the fresco.

**Comment by anurag-adhikari** [comment]
Looking forward to knowing more...

**Comment by itsabk** [comment]
was anyone able to find a work around?

**Comment by divyaswormakai** [comment]
I haven't checked the library on an app itself, but I can only see use cases where it is taking images from the device. Can it handle images that needs to be fetched from the cloud?

**Comment by Lakston** [comment]
How is this issue closed ? the base functionnality just does not work, I just had to debug for an hour to find out that `Image.getSize` basically never worked in react native

**Comment by divyaswormakai** [comment]
They closed it due to no activity on the issue.

**Comment by Lakston** [comment]
Well thanks for opening the issue in the first place (and the person who also openend one in 2018), at least I found out pretty quickly that the base functionnality of the Image component just never worked properly

**Comment by ghost1face** [comment]
I'm curious as to why there's no fix for this

**Comment by HarrisonChanPIC** [comment]
It's absolutely ridiculous that you can't even get the size of an image. Still. After over 6-7 years.

**Comment by Lakston** [comment]
From what I understood in my investigation, it's because they are using Fresco under the hood on Android and when Fresco measures an image it will take pixel density into account, I dont know if newer versions of Fresco have a method to avoid using pixel density and get the real size, or if there is an alternative to Fresco (although since Fresco is a Facebook lib I doubt they would switch), but this is really annoying, especially because it is not documented and clearly never worked properly since the Fresco update in 2018.

**Comment by jguerena15** [comment]
Just ran into this issue as well. For anyone who needs a work-around, use [Skia's useImage](https://shopify.github.io/react-native-skia/docs/images/) and you can grab width and height like:

```
const image = useImage(photoUri);
const width = image?.width();
const height = image?.height();
```

Annoying to have to use that if you're not already using Skia (I happened to be) but it works.

**Comment by Lakston** [comment]
As an other alternative, `getImageMetaData` from `react-native-compressor` also gives the correct dimensions, would love to see a fix in react native though and not have to rely on 3rd parties to do this.

**Comment by jhtr** [comment]
Apparently a fix was released in [v0.86.0](https://github.com/react/react-native/releases/tag/v0.86.0):

> Image: Image.getSize and Image.getSizeWithHeaders now return the true source dimensions instead of Fresco's downsampled values ([8823a8009b](https://github.com/facebook/react-native/commit/8823a8009b18950d5be2682a86cfcf0c7014a068) by [@zoontek](https://github.com/zoontek))
